src/inmanta/env.py

```diff
@@ -1182,13 +1183,28 @@ def mock_process_env(*, python_path: Optional[str] = None, env_path: Optional[st
     Overrides the process environment information. This forcefully sets the environment that is recognized as the outer Python
     environment. This function should only be called when a Python environment has been set up dynamically and this environment
     should be treated as if this process was spawned from it, and even then with great care.
-
     :param python_path: The path to the python binary. Only one of `python_path` and `env_path` should be set.
     :param env_path: The path to the python environment directory. Only one of `python_path` and `env_path` should be set.
+
+    When using this method in a fixture to set and reset virtualenv, it is preferable to use `store_venv()`
     """
     process_env.__init__(python_path=python_path, env_path=env_path)  # type: ignore
 
 
+def swap_process_env(env: ActiveEnv) -> ActiveEnv:
```

Comment:
I think this will be OK, but I'd suggest not mergint it on the same day as the pkg_resources PR so we can identify fallout.